core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -131,6 +132,9 @@ private FileIO io(String location) {
     String impl = implFromLocation(location);
     FileIO io = ioInstances.get(impl);
     if (io != null) {
+      if (io instanceof HadoopFileIO && ((HadoopFileIO) io).conf() == null) {
```

Comment:
This cannot modify the IO when accessing it. The underlying IO should use a `SerializableConfiguration` anyway, which is created in `setConf`. Why did you need to add this?